packages/mongodb-memory-server-core/src/util/getos/index.ts

```diff
@@ -42,21 +42,28 @@ export function isLinuxOS(os: AnyOS): os is LinuxOS {
   return os.os === 'linux';
 }
 
+/**
+ * Cache the "getOS" call, so that not much has to be re-executed over and over
+ */
+let cachedOs: AnyOS | undefined;
+
 /** Get an OS object */
 export async function getOS(): Promise<AnyOS> {
-  /** Node builtin function for first determinations */
-  const osName = platform();
-
-  // Linux is a special case.
-  if (osName === 'linux') {
-    return await getLinuxInformation();
+  if (!cachedOs) {
+    /** Node builtin function for first determinations */
+    const osName = platform();
+
+    // Linux is a special case.
+    if (osName === 'linux') {
+      cachedOs = await getLinuxInformation();
+    } else {
+      cachedOs = { os: osName };
+    }
   }
 
-  return { os: osName };
+  return cachedOs;
```

Comment:
why change the cache to be in getos instead of leaving it where it was?

Reply:
Since getOs is called in other files as well, I thought it would be more appropriate to cache it here.